### DISCUSSION TITLE
animated widget transitions

### DISCUSSION THREAD

**Discussion by nylki** [can-lehmann/owlkettle#154] [Q&A] [answered]
Hi there,
First of all, great library, I am having great fun with it! :)

I was wondering if animated transitions of views/widgets are already supported.
For example animating the hiding/showing between these two via a fadeIn/fadeOut:

```.nim
            case app.model.myEnum:
              of foo:
                Label(text = "foo")
              of bar:
                Label(text = "bar") 
```

I am not that familiar with GTK yet, but from the docs it seems like a [GtkRevealer](https://docs.gtk.org/gtk4/class.Revealer.html) could be used under the hood for that. Or perhaps with a (custom) style class?
Is something like that already possible with owlkettle?

**Comment by can-lehmann** [comment, marked as answer]
There are already some widgets that support animations, such as the Action Bar and Flap widgets. However I don't think we have a widget for animating transitions between multiple widgets yet. Usually, [Gtk.Stack](https://docs.gtk.org/gtk4/class.Stack.html) is used to animate transitions between multiple widgets. There is a [pull request](https://github.com/can-lehmann/owlkettle/pull/124) for adding support for Gtk.Stack, however it has yet to be reviewed.

**Reply by nylki** [reply]
Thank you for the overview, much appreciated! :+1:
